.github/workflows/docker-testbed.yml

```diff
@@ -0,0 +1,130 @@
+concurrency:
+  group: docker-testbed-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  docker-testbed:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # pin v4.1.1
+        timeout-minutes: 5
+
+      # Set up BuildKit Docker container builder to be able to build
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+      - name: Log in to the Github Container registry
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get current version
+        id: version
+        run: |
+          sed -n 's/^.*Tool.Parsec: "\(.*\)",$/current=\1/p' misc/version_updater.py | tee $GITHUB_OUTPUT
+
+      - name: Generate build metadata
+        uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
+        id: metadata
+        with:
+          images:
+            ghcr.io/scille/parsec-cloud/parsec-testbed-server
+          tags: |
+            type=raw,value=${{ steps.version.outputs.current }}+{{ date 'YYYYMMDD' }}.sha.{{ sha }}
+          flavor: |
+            latest=${{ github.event_name == 'workflow_dispatch' }}
+
+      - name: Build and export to Docker
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
+        id: build
+        with:
+          context: .
+          file: server/packaging/testbed-server/testbed-server.dockerfile
+          load: true
+          tags: ${{ steps.metadata.outputs.tags }}
+          labels: ${{ steps.metadata.outputs.labels }}
+          push: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Start docker test container
+        id: test-container
+        run: |
+          (
+            echo -n "id=";
+            docker run --detach --publish 6777:6777 --rm --name=parsec-testbed-server ${{ steps.build.outputs.imageid }}
+          ) | tee $GITHUB_OUTPUT
+
+      - name: Test docker image
+        run: |
+          import time
+          from urllib.request import Request, urlopen
+
+          r = Request('http://127.0.0.1:6777/testbed/new/empty', method='POST')
+          for i in range(10):
+            try:
+              urlopen(r)
+            except Exception as exc:
+              print(f'Try {i + 1}/10: {exc}')
+              time.sleep(1)
+              continue
+            else:
+              print('Success')
+              break
+          else:
+            raise SystemExit('Cannot connect to testbed server :(')
```

Comment:
Can we move this script to a file next to the .yml? I know it only tests the docker image but we may want to improve/update it independently of the workflow.

Reply:
done